aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/BaseHandlerStd.java

```diff
@@ -0,0 +1,213 @@
+package software.amazon.rds.dbinstance;
+
+import java.time.Duration;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import com.amazonaws.util.CollectionUtils;
+import software.amazon.awssdk.services.rds.RdsClient;
+import software.amazon.awssdk.services.rds.model.DBInstance;
+import software.amazon.awssdk.services.rds.model.DbInstanceNotFoundException;
+import software.amazon.awssdk.services.rds.model.DescribeDbInstancesRequest;
+import software.amazon.awssdk.services.rds.model.DescribeDbInstancesResponse;
+import software.amazon.awssdk.services.rds.model.ListTagsForResourceResponse;
+import software.amazon.awssdk.utils.StringUtils;
+import software.amazon.cloudformation.exceptions.CfnNotFoundException;
+import software.amazon.cloudformation.exceptions.CfnNotStabilizedException;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+import software.amazon.cloudformation.proxy.delay.Constant;
+
+public abstract class BaseHandlerStd extends BaseHandler<CallbackContext> {
```

Comment:
Should we gather all error handling here and process expected exception?
I think that will be easy to maintain and make cloud formation more aware of exceptions for any future feature [Like giving insights to customer(especially with automation scripts) for throttled requests or invalid parameter exceptions] and customer still  RDS exception anyway.

For sure It is not mandatory and code will work without it. But is it nice to have?

To relate, please take a look how `aws ApiGateway resource`  do it: https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-apigateway/blob/5751e9513c14c1248f1b3534efb5d7358887c837/resource/src/main/java/software/amazon/apigateway/resource/BaseHandlerStd.java#L58

Reply:
Good point, got it implemented here: https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-rds/pull/46/files#diff-3429914b497911f240a27f529350ff58bb2e18db96883b9c13f609194ae24c57R217-R241